pkg/infra/persistence/elasticsearch/client.go

```diff
@@ -417,3 +417,22 @@ func (cl *Client) termsAgg(ctx context.Context, index string, field string) (*Ag
 		Total:   len(bs),
 	}, nil
 }
+
+// PingElasticSearchClient would ping ElasticSearch client
+func (cl *Client) PingElasticSearchClient(ctx context.Context) error {
+	req := esapi.PingRequest{
+		Pretty: true,
+	}
+
+	// 使用带有上下文的请求
+	res, err := req.Do(ctx, cl.client)
```

Comment:
You can use `es.client.Info()` and `es.client.Cluster.Health()` instead directly, example:
```go
func checkElasticsearchLiveness(es *elasticsearch.Client) error {
    // 检查连接
    if _, err := es.Info() // 使用 Info 方法获取服务信息
    {
        return fmt.Errorf("failed to connect to Elasticsearch")
    }

    // 检查集群健康状态
    resp, err := es.Cluster.Health(nil)
    if err != nil {
        return fmt.Errorf("failed to get cluster health: %v", err)
    }
    defer resp.Body.Close()

    if status := resp.IsStatus("green", "yellow"); !status {
        return fmt.Errorf("Elasticsearch cluster health is not green or yellow")
    }

    return nil
}
```